app/controllers/hosts_controller.rb

```diff
@@ -629,6 +629,18 @@ def preview_host_collection
     end
   end
 
+  def get_statuses
+    statuses = []
+    HostStatus.status_registry.each do |status_class|
```

Comment:
I think `@host.host_statuses` gives you all the statuses for given host, so that could save you the check for relevancy and you could also call `to_status` and `to_label` directly on each status without the need for going meta with `send`.

Reply:
Shouldn't we keep the registry and in fact return also substatuses the host doesn't have value for? So we'd always get consistent set and for some statuses we'd say "unknown"?